src/sentry/static/sentry/app/components/contextPickerModal.jsx

```diff
@@ -0,0 +1,255 @@
+     */
+    nextPath: PropTypes.string.isRequired,
+
+    /**
+     * Finish callback
+     */
+    onFinish: PropTypes.func.isRequired,
+
+    /**
+     * Container for modal header
+     */
+    Header: PropTypes.oneOfType([PropTypes.element, PropTypes.func, PropTypes.string]),
+
+    /**
+     * Container for modal body
+     */
+    Body: PropTypes.oneOfType([PropTypes.element, PropTypes.func, PropTypes.string]),
+
+    /**
+     * List of available organizations
+     */
+    organizations: PropTypes.arrayOf(SentryTypes.Organization),
+
+    /**
+     * LatestContext store
+     */
+    latestContext: PropTypes.shape({
+      organization: SentryTypes.Organization,
+    }),
+
+    /**
+     * Does modal need to prompt for organization.
+     * TODO(billy): This can be derived from `nextPath`
+     */
+    needOrg: PropTypes.bool,
+
+    /**
+     * Does modal need to prompt for project
+     */
+    needProject: PropTypes.bool,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: props.organizations.length === 1,
+    };
+  }
+
+  componentDidMount() {
+    let {latestContext, organizations} = this.props;
+
+    // Don't make any assumptions if there are multiple organizations
+    if (organizations.length !== 1) {
+      return;
+    }
+
+    // If there is an org in context (and there's only 1 org available),
+    // attempt to see if we need more info from user and redirect otherwise
+    if (latestContext.organization) {
+      // This will handle if we can intelligently move the user forward
+      this.checkShouldFinish(
+        [latestContext.organization],
+        latestContext.organization.projects
+      );
+      return;
+    }
+
+    // Since user belongs to only 1 org, we can set it as active and componentWillReceiveProps handle the rest
+    fetchOrganizationDetails(organizations[0].slug, {
+      setActive: true,
+      loadProjects: true,
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // Should only check the case where there is no latestContext.organization and we're waiting
+    // for it to be set (after fetch in DidMount)
+    let {latestContext} = this.props;
+    if (
+      (!latestContext.organization &&
+        latestContext.organization !== nextProps.latestContext.organization) ||
+      (latestContext.organization &&
+        nextProps.latestContext.organization &&
+        latestContext.organization.slug !== nextProps.latestContext.organization.slug)
+    ) {
+      // Check if we can push the user forward w/o needing them to select anything
+      this.checkShouldFinish(
+        this.props.organizations,
+        nextProps.latestContext.organization.projects,
+        nextProps.latestContext.organization
+      );
+    }
+  }
+
+  // Performs checks to see if we need to prompt user
+  // i.e. When there is only 1 org and no project is needed or
+  // there is only 1 org and only 1 project (which should be rare)
+  checkShouldFinish = (
```

Comment:
I'm not sure I'd expect a method named `checkShouldFinish` to have a side effect of redirecting/navigating somewhere. How about `redirectIfFinished` or `navigateIfFinished`?